doc/design_ros_2.md

```diff
@@ -347,7 +394,7 @@ Nodes are instanciated, usually as a `shared_ptr` through `std::make_shared<Node
 After all the nodes have been added, `rclcpp::Executor::spin()` is called (there are other spinning varations, but this is the main one).
 `rclcpp::executors::SingleThreadedExecutor::spin()` simply loops forever until the process' context isn't valid anymore.
 It fetches the next `rclcpp::AnyExecutable` (e.g., subscription, timer, service, client), possibly waiting a bit, and calls `rclcpp::Executor::execute_any_executable()` with it.
-This then calls the relevant `execute*()` method (e.g., [`execute_timer()`](#timer-callbacks), [`execute_subscription()`](#subscription-callbacks), `execute_service()`, `execute_client()`).
+This then calls the relevant `execute*()` method (e.g., [`execute_timer()`](#timer-callbacks), [`execute_subscription()`](#subscription-callbacks), `execute_service()`, `execute_client()`, `execute_any_executable()`).
```

Comment:
I think what you mean here to add is `Waitable` entity to execute?

```suggestion
This then calls the relevant `execute*()` method (e.g., [`execute_timer()`](#timer-callbacks), [`execute_subscription()`](#subscription-callbacks), `execute_service()`, `execute_client()`, `Waitable::execute()`).
```

Reply:
Yeah, this makes more sense, especially if you look at `rclcpp::Executor::execute_any_executable()` and see that all these functions (can) get called.

https://github.com/ros2/rclcpp/blob/3088b536cc02f0e5d01aa4bccda0cc2d2d8c1ebf/rclcpp/src/rclcpp/executor.cpp#L522-L559

Reply:
I applied the fix